hack/podman-registry-go/registry_test.go

```diff
@@ -0,0 +1,31 @@
+package registry
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartAndStopMultipleRegistries(t *testing.T) {
+	binary = "../podman-registry"
+
+	registries := []*Registry{}
+
+	// Start registries.
+	for i := 0; i < 3; i++ {
+		reg, err := Start()
+		require.NoError(t, err)
+		assert.True(t, len(reg.Image) > 0)
+		assert.True(t, len(reg.User) > 0)
+		assert.True(t, len(reg.Password) > 0)
+		assert.True(t, len(reg.Port) > 0)
+		registries = append(registries, reg)
+	}
+
+	// Stop registries.
+	for _, reg := range registries {
```

Comment:
Would it make sense to add a `teardown` that repeats this loop, but without error checks? My concern is making sure we clean up even in the case of an assertion failure.

Bonus points for doing it cleanly, e.g. by removing from `registries` in the loop below, or by having `reg.Stop()` set an already-stopped flag, etc.

Reply:
I changed it to use a multierror. Good catch!

The registry is now also carrying an internal state field to know it was already stopped.